[Li Fei] iP

LF1928 opened the pull request:
# Hello
> This is a quote
## Section 1
* Task 1
   - by [Monday](https://nus-cs2103-ay2425s2.github.io/website/schedule/week4/project.html)
   - 7pm
* ~~Task 2~~
* Task 3
1. Most urgent
2. 2nd most
3. least urgent
`markAsDone` method
``` java
public static void markAsDone(ArrayList<Task> listOfTasks, int taskNumber) throws InvalidTaskNumberException {
    checkIfTaskIsValid(listOfTasks, taskNumber);
    listOfTasks.get(taskNumber - 1).markAsDone();
}
```
## Features to **add**
- [x] Find
- [ ] Order based on deadline
- [ ] Add delight to the experience when all tasks are complete :tada:

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

LF1928 committed:
Rename main class and logo

LF1928 committed:
Add greeting

LF1928 committed:
Add support for echo

LF1928 committed:
Add suport for list and add

LF1928 committed:
Added Task class and support for Mark as done and undone

LF1928 committed:
Added subclasses for different tasks

LF1928 committed:
add support for different types of tasks

LF1928 committed:
added Automated Text UI Testing

LF1928 committed:
add exceptions for missing description

LF1928 committed:
add support for delete

LF1928 committed:
Add enum and reorganise

LF1928 committed:
Add support for load file and store task

LF1928 committed:
Remove debugging messages

LF1928 committed:
Add support for update after conversation

LF1928 committed:
Merge branch 'branch-Level-7'

LF1928 committed:
Add support for date and time

LF1928 committed:
Merge branch 'branch-Level-8'

LF1928 committed:
Add more OOP

LF1928 committed:
Merge branch 'MoreOOP'

LF1928 committed:
Add packages

LF1928 committed:
Merge branch 'packages'

LF1928 committed:
Merge branch 'add-gradle-support'

LF1928 committed:
Add JUnit Tests

LF1928 committed:
Add JavaDoc comments to the code

LF1928 committed:
Comply to coding standard

LF1928 committed:
Add support for find

LF1928 committed:
Merge branch 'branch-A-JavaDoc'

LF1928 committed:
Merge branch 'branch-A-CodingStandard'

LF1928 committed:
Merge branch 'branch-Level-9'

LF1928 committed:
Match coding standards

LF1928 committed:
Add support for GUI v1

LF1928 committed:
Adjust display for GUI

LF1928 committed:
Merge branch 'branch-Level-10'

xk1234 commented on src/main/java/duke/parsers/DateTimeParse.java:
```diff
@@ -0,0 +1,104 @@
+ * date-time strings from commands, parse them into {@link LocalDateTime} or {@link LocalDate},
+ * and format them into user-friendly strings.
+ */
+public class DateTimeParse {
```
Would you consider whether something like DateTimeParser might feel more like a noun than DateTimeParse?

xk1234 commented on src/main/java/duke/parsers/Parser.java:
```diff
@@ -0,0 +1,106 @@
+     */
+    public static String extractKeyword(String userInput) {
+        return userInput.substring(5).trim();
+    }
```
I like how your method names extractKeyword give a clear sense of what the method does.

xk1234 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,94 @@
+        public String execute(String input, ArrayList<Task> listOfTasks) {
+            return Ui.bye();
+        }
+    },
```
I like that your enum Command is all uppercase constants (BYE, LIST, MARK), which is standard for enums.

xk1234 commented on src/test/java/DateTimeParseTest.java:
```diff
@@ -0,0 +1,53 @@
+            DateTimeParse.parseDateTime("Invalid Date Format");
+        });
+        assertEquals("Invalid date-time format: Invalid Date Format", exception.getMessage());
+    }
```
For your tests (testParseDateTime_invalidFormat_expectedBehavior), I like that you’re following a descriptive pattern: it tells me exactly which scenario you’re testing and what behavior you expect.

xk1234 commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,129 @@
+        }
+    }
+
+    private void handleUserInput() {
```
I like how you named methods such as handleUserInput. I think the words are in a sensible order, and it’s clear what the method does.

xk1234 reviewed (commented):
LGTM. 1-2 naming changes recommended

LF1928 committed:
edit GUI

LF1928 committed:
trim Main and DialogBox

LF1928 committed:
Merge branch 'GUI-edits'

LF1928 committed:
Fix bugs in initialise

LF1928 committed:
Reorganize methods in main

LF1928 committed:
Merge branch 'master' into branch-A-CodeQuality

LF1928 committed:
Merge pull request #1 from LF1928/branch-A-CodeQuality

Reorganize methods in main

LF1928 committed:
Add assertion

LF1928 committed:
Add assertion

LF1928 committed:
Merge pull request #2 from LF1928/master

Add assertion

LF1928 committed:
Merge branch 'branch-A-Assertions'

LF1928 committed:
Merge branch 'branch-A-Assertions'

LF1928 committed:
Add support for Update

Update only for description of the specified task
Update Command is not case sensitive
Update description is applicable to all types of Tasks

Input format only update {taskNumber} {newTaskDescription}

LF1928 committed:
Add support for clone

Clone can duplicate the task based on index number
Clone is case-sensitive

All types of tasks can be cloned

LF1928 committed:
Merge branch 'branch-Extension'

LF1928 committed:
Update name of chatBot

LF1928 committed:
Add error handling for DateParseException and IndexOutOfBoundsException

LF1928 committed:
Merge branch 'branch-A-MoreErrorHandling'

LF1928 committed:
Align the text box to the right for the user

LF1928 committed:
Improve GUI by changing wallpaper

LF1928 committed:
Merge branch 'branch-A-BetterGui'

LF1928 committed:
Align with Coding standard

LF1928 committed:
Add a user guide for the project

LF1928 committed:
Add testing for extractNewTaskDescription

LF1928 committed:
Merge branch 'branch-A-MoreTesting'

LF1928 committed:
Update display of text

LF1928 committed:
Merge branch 'branch-ReorganizeUi'

LF1928 committed:
Update User Guide

LF1928 committed:
Update name of Chatbot

LF1928 committed:
Update name of doc

LF1928 committed:
Handle more errors

LF1928 committed:
Update date instructions

LF1928 committed:
Update for dead code
